src/libraries/System.Xml.XDocument/ref/System.Xml.XDocument.cs

```diff
@@ -199,45 +212,45 @@ public partial class XDocument : System.Xml.Linq.XContainer
     }
     public partial class XDocumentType : System.Xml.Linq.XNode
     {
-        public XDocumentType(string name, string publicId, string systemId, string internalSubset) { }
+        public XDocumentType(string name, string? publicId, string? systemId, string internalSubset) { }
```

Comment:
double check if internalSubset should be non-null

Reply:
I wrote a TODO explaining why this was non-nullable when I annotated the src:
https://github.com/dotnet/runtime/blob/7408f1b0e4a184ff24dd1a48bd74dbb51aad7e4f/src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XDocumentType.cs#L56-L64

But given that homonymous property in `XmlDocumentType` is nullablle I am willing to make this nullable... However doing that involves changing XNodeReader.Value to `string?`.
https://github.com/dotnet/runtime/blob/7408f1b0e4a184ff24dd1a48bd74dbb51aad7e4f/src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XNodeReader.cs#L358

And that last thing involves changing the definition of `XmlReader.Value` to `string?` since that's the base type and this last thing which will potentially trigger a lot of nullability changes.

Reply:
File an issue on this and leave as is

Reply:
actually, TODO-NULLABLE is fine

Reply:
(resolving)